src/content/en/fundamentals/getting-started/primers/async-functions.md

```diff
@@ -0,0 +1,345 @@
+
+    async function logFetch(url) {
+      try {
+        const response = await fetch(url);
+        console.log(await response.text());
+      }
+      catch (err) {
+        console.log('fetch failed', err);
+      }
+    }
+
+It's the same number of lines, but all the callbacks are gone. This makes it way
+easier to read, especially for those less familiar with promises.
+
+## Other async function syntax
+
+We've seen `async function() {}` already, but the `async` keyword can be used
+with other function syntax:
+
+### Arrow functions
+
+    // map some URLs to json-promises
+    const jsonPromises = urls.map(async url => {
+      const response = await fetch(url);
+      return response.json();
+    });
+
+Note: `array.map(func)` doesn't care that I gave it an async function, it just
+sees it as a function that returns a promise. It won't wait for the first
+function to complete before calling the second.
+
+### Object methods
+
+    const storage = {
+      async getAvatar(name) {
+        const cache = await caches.open('avatars');
+        return cache.match(`/avatars/${name}.jpg`);
+      }
+    };
+
+    storage.getAvatar('jaffathecake').then(…);
+
+### Class methods
+
+    class Storage {
+      constructor() {
+        this.cachePromise = caches.open('avatars');
+      }
+
+      async getAvatar(name) {
+        const cache = await this.cachePromise;
+        return cache.match(`/avatars/${name}.jpg`);
+      }
+    }
+
+    const storage = new Storage();
+    storage.getAvatar('jaffathecake').then(…);
+
+Note: Class constructors cannot be async.
+
+## Example: Streaming a response
+
+The benefit of async functions increases in more complex examples. Say we wanted
+to stream a response while logging out the chunks and returning the final size.
+
+Note: The phrase "logging out the chunks" made me sick in my mouth.
+
+Here it is with promises:
+
+    function getResponseSize(url) {
+      return fetch(url).then(response => {
+        const reader = response.body.getReader();
+        let total = 0;
+
+        return reader.read().then(function processResult(result) {
+          if (result.done) return total;
+
+          const value = result.value;
+          total += value.length;
+          console.log('Received chunk', value);
+
+          return reader.read().then(processResult);
+        })
+      });
+    }
+
+Check me out, Jake "wielder of promises" Archibald. See how I'm calling
+`processResult` inside itself to set up an asynchronous loop? Writing that made
+me feel *very smart*. But like most "smart" code, you have to stare at it for
+ages to figure out what it's doing, like one of those magic-eye pictures from
+the 90's.
+
+Let's try that again with async functions:
+
+    async function getResponseSize(url) {
+      const response = await fetch(url);
+      const reader = response.body.getReader();
+      let result;
+      let total = 0;
+
+      while ((result = await reader.read()) && !result.done) {
```

Comment:
I like how this uses a loop, and how it uses a bunch of await calls, and seems really practical. However, maybe it would be easier to read if you didn't use the pattern of assignment `&&` another expression, instead breaking it out into another line (otherwise it still may have the smell of "wielder of something").